library/Makefile

```diff
@@ -199,15 +199,15 @@ endif
 
 libmbedtls.$(SOEXT_TLS): $(OBJS_TLS) libmbedx509.so
 	echo "  LD    $@"
-	$(CC) -shared -Wl,-soname,$@ -L. -lmbedcrypto -lmbedx509 $(LOCAL_LDFLAGS) $(LDFLAGS) -o $@ $(OBJS_TLS)
+	$(CC) -shared -Wl,-soname,$@ -o $@ $(OBJS_TLS) -L. -lmbedcrypto -lmbedx509$(LOCAL_LDFLAGS) $(LDFLAGS)
```

Comment:
Isn't it enough to link mbedtls.so to libmbedx509.so only? Given that libmbedx509.so is linked to libmbedcrypto.so.

Reply:
@haampie While you are swapping the linking order for `libmbedtls.so` and `libmbedcrypto.dylib`, could you do so for `libmbedcrypto.dll` as well?

The object files are already listed in the right place, but `-lmbedcrypto˙` and `-lmbedx509` need to be swapped in that rule as well.

Reply:
> Isn't it enough to link libmbedtls.so to libmbedx509.so only? 

It would work, but it wouldn't be technically correct: mbedtls code also calls crypto directly. In principle, there could be configurations where x509 doesn't call crypto (although I can't imagine a plausible one). So omitting libmbedcrypto would make the code harder to understand, and I don't see any advantage.